arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentCollectionViewController.swift

```diff
@@ -30,7 +30,8 @@ class ContentCollectionViewController: UIViewController, UICollectionViewDataSou
     
     private var headerView:CustomSearchHeaderView!
     
-    var nodesArray:[Node]!
+    /// The categories to display in the collection view.
+    var categories = [Category]()
```

Comment:
This variable is set twice. Once here and once in the App Delegate. Consider reducing this to once. As a result, this member could be made static.

Reply:
My goal was to remove the source (location) of the categories/samples as a dependency from `ContentCollectionViewController`. You are correct that the variable is set twice (once on initialization and once when it is given the data to prevent), but creating an empty array is a very inexpensive operation.

I could update the variable rather than assign a new array:
```swift
let categories = try PropertyListDecoder().decode([Category].self, from: data)
contentCollectionViewController.categories.append(contentsOf: categories)
```
But I'm not sure that makes any practical difference.

Reply:
Following your goal, there's also the of declaring categories as an implicit unwrapped optional:

```swift
var categories: [Category]!
```